src/App.tsx

```diff
@@ -1,16 +1,29 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const addMovie = (movie: Movie) => {
+    const newMovie = {
+      ...movie,
+    };
+
+    setMovies(currentMovies => [...currentMovies, newMovie]);
+  };
```

Comment:
```suggestion
  const addMovie = (movie: Movie) => {
    setMovies(currentMovies => [...currentMovies, movie]);
  };
```

can be a bit simplified in this case